engines/suppliers/app/presenters/suppliers/producer_presenter.rb

```diff
@@ -0,0 +1,44 @@
+module Suppliers
+  class ProducerPresenter
+    # @param producer [Producer]
+    # @param user [User]
+    def initialize(producer, user)
+      @producer = producer
+      @user = user
+    end
+
+    # Checks if the given user has edit rights on the producer
+    #
+    # @return [Boolean]
+    def can_edit
+      producer.has_admin?(user) || group_admin?
+    end
+
+    delegate :id, :name, :email, :address, :created_at, :updated_at, to: :producer
```

Comment:
Sure, this is just a proptotype. We may drop the presenter and flag thing all the way.